src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.cs

```diff
@@ -33,7 +33,7 @@ public static partial class JsonSerializer
 
                         if (HandleValue(tokenType, options, ref reader, ref state))
```

Comment:
As an aside, @steveharter, returning "true" in these `HandleX` helper methods to indicate there is no more data/we are done is the opposite of what I would have expected. We should consider inverting all these conditions so that "true" means success/keep reading, and false means "i'm done/return to the caller".

Reply:
Provided I can address all of your other feedback, I'm up for making this change (unless it would be better done in a follow-up PR).

Reply:
> Provided I can address all of your other feedback, I'm up for making this change

Great! cc @JeremyKuhne (in case you disagree with flipping the condition).

> unless it would be better done in a follow-up PR

I think that would be best. Let's keep this PR isolated to the bug fix.

Reply:
Sounds good. I'll wait until the other PR's land that are affecting HandleXXX.

Reply:
One benefit of the direct return like the original is to support future cases (in the upcoming extensibility model) where the reader may be passed into a new static serializer `Parse()` method. In that case we want to stop when we finish reading the current scope (object for example) and not throw an exception. There may be different ways to detect\support that but it is something to track when we add that feature.

Reply:
Does JsonDocument support trailing trivia?

Reply:
Yes, it does. We have a few tests around that:
https://github.com/dotnet/corefx/blob/10d0327c8d7be6e6cd743ddbcc1d74e329fc2360/src/System.Text.Json/tests/JsonDocumentTests.cs#L783
https://github.com/dotnet/corefx/blob/10d0327c8d7be6e6cd743ddbcc1d74e329fc2360/src/System.Text.Json/tests/JsonDocumentTests.cs#L1447

cc @bartonjs